domehunter/gRPC-server/hx2dome.proto_server.py

```diff
@@ -0,0 +1,578 @@
+
+        Returns
+        -------
+        ReturnCode
+            rpc response, a message of type 'ReturnCode' that indicates the
+            success/failure of the rpc request.
+
+        """
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            # TODO: unsure what action to trigger in automationHAT for "Park"
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+
+    def dapiUnpark(self, request, context):
+        """TheSkyX RPC to unpark the dome.
+        NOTE: this functionality isn't currently implemented.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        ReturnCode
+            rpc response, a message of type 'ReturnCode' that indicates the
+            success/failure of the rpc request.
+
+        """
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            # TODO: unsure what action to trigger in automationHAT for "UnPark"
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+
+    def dapiFindHome(self, request, context):
+        """TheSkyX RPC to return dome to the home position.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        ReturnCode
+            rpc response, a message of type 'ReturnCode' that indicates the
+            success/failure of the rpc request.
+
+        """
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            return_code = 0
+            try:
+                self.dome.find_home()
+            except:
+                # TODO: proper error handling
+                return_code = 1
+            response = hx2dome_pb2.ReturnCode(return_code=return_code)
+            return response
+
+    def dapiIsGotoComplete(self, request, context):
+        """TheSkyX RPC to request the completetion status of dapiGotoAzEl.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        IsComplete
+            rpc response, a message of type 'IsComplete' containing a
+            return_code that indicates the success/failure of the rpc request.
+            and a boolean is_complete that indicates if the relevant rpc
+            has been completed.
+
+        """
+        print("      ITS DOING IT, ITS DOING IT    ")
```

Comment:
It's alive!!!  But should probably make this more helpful.

Reply:
Yea it just made it a bit more interesting when I was trying to get things to work at all. Will replace all this when I do logging in future pr